lua/Network.lua

```diff
@@ -0,0 +1,264 @@
+
+	for i = 1, nPoint - 1 do
+		endPoint = binding.te.gm.Point(geometry:getX(i), geometry:getY(i), geometry:getSRID())
+	end
+
+	local ffPointer = {p1 = beginPoint, p2 = endPoint}
+
+	return ffPointer
+end
+
+local function distancePoint(lines, target)
+	local arrayTargetLine = {}
+	local counter = 0
+
+	forEachCell(target, function(targetPoint)
+		local geometry1 = tl:castGeomToSubtype(targetPoint.geom:getGeometryN(0))
+		local nPoint1 = geometry1:getNPoints()
+		local distance
+		local minDistance = math.huge
+		local point
+
+		forEachCell(lines, function(line)
+			local geometry2 = tl:castGeomToSubtype(line.geom:getGeometryN(0))
+			local nPoint2 = geometry2:getNPoints()
+
+			for i = 0, nPoint2 do
+				point = tl:castGeomToSubtype(geometry2:getPointN(i))
+				distance = geometry1:distance(point)
+
+				if distance < minDistance then
+					minDistance = distance
+					arrayTargetLine[counter] = point
+				end
+			end
+		end)
+
+		if arrayTargetLine[counter] ~= nil then
+			counter = counter + 1
+		end
+	end)
+
+	return arrayTargetLine
+end
+
+local function checksInterconnectedNetwork(data)
+	local ncellRed = 0
+
+	forEachCell(data.lines, function(cellRed)
+		local bePoint = getBELine(cellRed)
+		local lineValidates = false
+		local differance = math.huge
+		local distance = 20
+		local redPoint
+
+		for j = 0, 1 do
+			if j == 0 then
+				redPoint = tl:castGeomToSubtype(bePoint.p1)
+			else
+				redPoint = tl:castGeomToSubtype(bePoint.p2)
+			end
+
+			local ncellBlue = 0
+
+			forEachCell(data.lines, function(cellBlue)
+				local geometry = tl:castGeomToSubtype(cellBlue.geom:getGeometryN(0))
+				local nPoint = geometry:getNPoints()
+
+				for i = 0, nPoint do
+					local bluePoint = tl:castGeomToSubtype(geometry:getPointN(i))
+
+					if ncellRed ~= ncellBlue then
+						distance = redPoint:distance(bluePoint)
+
+						if distance <= data.error then
+							lineValidates = true
+						end
+
+						if differance > distance then
+							differance = distance
+						end
+					end
+				end
+
+				if lineValidates == true then
+					return false
+				end
+
+				ncellBlue = ncellBlue + 1
+			end)
+		end
+
+		if not lineValidates then
+			customError("line do not touch, They have a differance of: "..differance)
+		end
+
+		ncellRed = ncellRed + 1
+	end)
+end
+
+Network_ = {
+	type_ = "Network",
+	--- Creates and validates a network.
+	-- @arg data.target File with the path of network end points.
+	-- @arg data.lines File with the path of a network.
+	-- @usage csCenterspt = CellularSpace{
+	--	file = filePath("rondonia_urban_centers_pt.shp", "gpm")
+	--}
+	--local csLine = CellularSpace{
+	--	file = filePath("rondonia_roads_lin.shp", "gpm")
+	--}
+	--local nt = Network{
+	--	target = csCenterspt,
+	--	lines = csLine
+	--}
+	--nt:createOpenNetwork{
+	--	target = filePath("rondonia_urban_centers_pt.shp", "gpm"),
+	--	lines = filePath("rondonia_roads_lin.shp", "gpm")
+	--}
+	createOpenNetwork = function(self, data)
```

Comment:
This function should be local and outside `Network_`. It must be automatically called within function `Network`.